docs/examples.md

```diff
@@ -420,6 +420,115 @@ object PerftConverter extends IOApp.Simple {
 ```
 @:@
 
+## Writing data to a CSV file
+
+If you want to save a list of a case class into a CSV file this helper method may aid you:
+
+@:select(scala-version)
+
+@:choice(scala-3)
+```scala mdoc:reset:silent
+//> using lib "org.typelevel::toolkit::@VERSION@"
+
+import fs2.data.csv.*
+import fs2.data.csv.generic.semiauto.*
+import fs2.io.file.{Files, Path}
+import cats.effect.{IO, IOApp}
+import fs2.{Pipe, Stream}
+
+/** Define your case class and derive an encoder for it */
+case class YourCaseClass(n: String, i: Int)
+given CsvRowEncoder[YourCaseClass, String] = deriveCsvRowEncoder
+
+/** This is our helper function that writes a case class as a csv given a path. */
+def writeCaseClassToCsv[A](
+    path: Path
+)(using CsvRowEncoder[A, String]): Pipe[IO, A, Nothing] =
+  _.through(encodeUsingFirstHeaders(fullRows = true))
+    .through(fs2.text.utf8.encode)
+    .through(Files[IO].writeAll(path))
+    .drain
```

Comment:
`.drain` here is redundant. `Files[IO].writeAll(path)` already return a `Stream[IO, Nothing]`